backend/src/mirrors_qa_backend/db/mirrors.py

```diff
@@ -1,140 +1,119 @@
-from typing import Any
-from urllib.parse import urlsplit
+from dataclasses import dataclass
 
-import requests
-from bs4 import BeautifulSoup, NavigableString
-from bs4.element import Tag
 from sqlalchemy import select
 from sqlalchemy.orm import Session as OrmSession
 from sqlalchemy.orm import selectinload
 
 from mirrors_qa_backend import logger, schemas
 from mirrors_qa_backend.db import models
-from mirrors_qa_backend.settings import Settings
+from mirrors_qa_backend.exceptions import EmptyMirrorsError
 
 
-def create_mirrors(session: OrmSession, countries: list[schemas.Country]) -> None:
-    for country in countries:
-        c = models.Country(code=country.code, name=country.name)
-        c.mirrors = [models.Mirror(**m.model_dump()) for m in country.mirrors]
-        session.add(c)
+@dataclass
+class UpdateMirrorsResult:
+    """Represents the results of an update to the list of mirrors in the database"""
 
+    nb_mirrors_added: int = 0
+    nb_mirrors_disabled: int = 0
 
-def update_mirrors(session: OrmSession, countries: list[schemas.Country]) -> None:
+
+def create_mirrors(session: OrmSession, mirrors: list[schemas.Mirror]) -> int:
+    """
+    Saves all the mirrors to the database.
+
+    Returns the total number of mirrors created.
+
+    Assumes that each mirror does not exist on the database.
     """
-    Updates the status of mirrors in the database. Any mirrors in the database
-    that do not exist in the current mirrors obtained from `countries` are
-    marked as disabled. New mirrors are saved accordingly.
+    total = 0
+    for mirror in mirrors:
+        db_mirror = models.Mirror(
+            id=mirror.id,
+            base_url=mirror.base_url,
+            enabled=mirror.enabled,
+            region=mirror.region,
+            asn=mirror.asn,
+            score=mirror.score,
+            latitude=mirror.latitude,
+            longitude=mirror.longitude,
+            country_only=mirror.country_only,
+            region_only=mirror.country_only,
+            as_only=mirror.as_only,
+            other_countries=mirror.other_countries,
+        )
+        # Ensure the country exists for the mirror
+        country = session.scalars(
+            select(models.Country).where(models.Country.code == mirror.country.code)
+        ).one_or_none()
+
+        if country is None:
+            country = models.Country(code=mirror.country.code, name=mirror.country.name)
+            session.add(country)
+
+        db_mirror.country = country
+        session.add(db_mirror)
+        logger.debug(
+            f"Registered new mirror: {db_mirror.id!r} for country: {country.name!r}"
+        )
+        total += 1
+    return total
+
+
+def update_mirrors(
```

Comment:
I think the name is not appropriate anymore given what it does:

- it creates mirror which does not exist
- it doesnt update mirror details (country) except for Enable property

Maybe `create_or_update_status()` is more informative.